packs/nest/actions/fan.yaml

```diff
@@ -0,0 +1,24 @@
+---
+name: fan
+runner_type: run-python
+description: Manage fan state
+enabled: true
+entry_point: fan.py
+parameters:
+  state:
+    type: string
+    description: State to set the fan
+    enum:
+      - "on"
+      - auto
+    default: auto
+    required: true
+  structure:
+    type: integer
+    description: ID of Nest Home name
+    default: 0
+  device:
```

Comment:
I usually combine both options together - function defaults and parameter defaults in the metadata file.

It might be slightly confusing to the pack author / contributor if the default values get out of sync, but for the end user it doesn't really make a difference since parameter defaults override function defaults.

Function defaults are there to make code more clear, reusable and useful if the action is used outside the StackStorm action context.